### ISSUE TITLE
"/usr/bin/kill: No such file or directory" in Docker.log when "Terminating com.docker.backend..."

### ISSUE THREAD

**Issue by pythiankerr** [docker/for-mac#6589]
- [x] I have tried with the latest version of Docker Desktop
  - [x] I have tried disabling enabled experimental features
  - [x] I have uploaded Diagnostics
  - Diagnostics ID: F37AEFC2-73B1-4F44-A83F-02FA5E1AC38C/20221121110657
  - Diagnostics ID: (newer) F37AEFC2-73B1-4F44-A83F-02FA5E1AC38C/20221128003318

### Expected behavior

Upon watchdog failure, Docker should be successfully terminated and then be relaunched.

### Actual behavior

The docker VM kernel suffered some fault and Docker needed to terminate it.

"Docker is stopping...." hangs indefinitly. Docker.log indicates it is using /usr/bin/kill to kill process 900 (com.docker.backend), but there is "No such file or directory"

```
time="2022-10-24T03:28:04Z" level="info" msg="GoBackend: Terminating com.docker.backend (89006) and waiting until exit"
time="2022-10-24T03:28:04Z" level="info" msg="GoBackend: Closing com.docker.backend wathdog pipe"
time="2022-10-24T03:28:04Z" level="info" msg="GoBackend: Waiting for com.docker.backend to finish."
time="2022-10-24T03:29:44Z" level="info" msg="GoBackend: com.docker.backend (89006) has still not exited, killing it."
time="2022-10-24T03:29:44Z" level="error" msg="Cmd: /bin/sh: /usr/bin/kill: No such file or directory
"
```

On my macOS /usr/bin/kill indeed does not exist; instead /bin/kill does. Using `kill -QUIT 900` immediately recovered the situation and I was able to successfully restart Docker.

### Information

You can see the path is present in the Docker binary:

```
$ strings /Applications/Docker.app/Contents/MacOS/Docker | grep /usr/bin/kill
/usr/bin/kill -9
```

Reproducible: perhaps, if you could cause the VM kernel to become unresponsive. I did notice that a 'node' process (presumably from VSCode Dev Container) was timing out (hung kernel task) buried within do_statx. I had been out and about working remotely for part of the day, but had been working happily for a couple of hours by the time it crashed.

New Problem?: I think I noticed increased instability for a few updates now.... perhaps a few months.

  - macOS Version: Monterey 12.6.1
  - Intel chip or Apple chip: Apple M1 Pro
  - Docker Desktop Version: 4.13.1 (90346)

### Output of `/Applications/Docker.app/Contents/MacOS/com.docker.diagnose check`

(no longer relevant, but at the the time it did detect that Docker wasn't running, and reported numerous EOF errors)

**Comment by pythiankerr** [comment]
Issue recurred; uploaded new diagnostics report.

I was using VS Code with Remote Containers at the time, not doing anything particular, just editing a file, when the connection to Docker was reported to have failed by VS Code.

The following was again observed in Docker.log

```
time="2022-11-28T00:33:23Z" level="info" msg="GoBackend: com.docker.backend (76220) has still not exited, killing it."
time="2022-11-28T00:34:01Z" level="error" msg="Cmd: /bin/sh: /usr/bin/kill: No such file or directory
```

Earliest sign of error is in ./log/host/com.docker.driver.amd64-linux.log

```
[2022-11-28T00:30:47.691025000Z][com.docker.driver.amd64-linux][I] reading from vpnkit: EOF
[2022-11-28T00:30:47.696229000Z][com.docker.driver.amd64-linux][W] Error closing raw stream to client: close unix <HOME>/.docker/run/docker.sock->: shutdown: socket is not connected
```

Last time this happened it happened on a Monday also.

I am now resetting (Docker) to factory defaults... although I had to kill the backend process to be able to do this from the GUI.

**Comment by pythiankerr** [comment]
Noticed there is an update available: updating from 4.13.1 (90346) to 4.14.1 (91661).

**Comment by pythiankerr** [comment]
To clarify, has this been actually fixed (if so, should I expect it in the next release?) or has it been closed for some other reason?

The problem is still there in 4.14.1

```
$ strings /Applications/Docker.app/Contents/MacOS/Docker | grep /usr/bin/kill
/usr/bin/kill -9
```

Thanks,
Cameron

**Comment by mat007** [comment]
A fix has been released in Docker Desktop 4.16. See the [release notes](https://docs.docker.com/desktop/release-notes/) for more details.
